app/battle/preparation.py

```diff
@@ -0,0 +1,21 @@
+def battle_preparation(knight: dict) -> None:
+    # BATTLE PREPARATIONS:
+
+    # apply armour
+    knight["protection"] = 0
+    for arm in knight["armour"]:
+        knight["protection"] += arm["protection"]
+
+    # apply weapon
+    knight["power"] += knight["weapon"]["power"]
```

Comment:
Ensure that `knight['power']` is initialized before adding the weapon's power. Consider setting `knight['power'] = 0` before this line to avoid potential errors if the power attribute is not initialized.